forc-plugins/forc-crypto/src/keys/vanity.rs

```diff
@@ -0,0 +1,469 @@
+use fuel_crypto::{fuel_types::Address, PublicKey, SecretKey};
+use fuels_accounts::wallet::{generate_mnemonic_phrase, DEFAULT_DERIVATION_PATH_PREFIX};
+use fuels_core::types::{
+    bech32::{Bech32Address, FUEL_BECH32_HRP},
+    checksum_address::checksum_encode,
+};
+use rayon::iter::{self, Either, ParallelIterator};
+use regex::Regex;
+use serde_json::json;
+use std::{
+    path::PathBuf,
+    time::{Duration, Instant},
+};
+use tokio::runtime::Runtime;
+
+forc_util::cli_examples! {
+    crate::Command {
+        [ Generate a checksummed vanity address with a given prefix => "forc crypto vanity --starts-with \"aaa\"" ]
+        [ Generate a checksummed vanity address with a given suffix => "forc crypto vanity --ends-with \"aaa\"" ]
+        [ Generate a checksummed vanity address with a given prefix and suffix => "forc crypto vanity --starts-with \"00\" --ends-with \"ff\"" ]
+        [ Generate a checksummed vanity address with a given regex pattern => "forc crypto vanity --regex \"^00.*ff$\"" ]
+    }
+}
+
+fn validate_hex_string(s: &str) -> Result<String, String> {
+    if !s.chars().all(|c| c.is_ascii_hexdigit()) {
+        return Err("Pattern must contain only hex characters (0-9, a-f)".to_string());
+    }
+    Ok(s.to_string())
+}
+
+fn validate_regex_pattern(s: &str) -> Result<String, String> {
+    if s.len() > 128 {
+        return Err("Regex pattern too long: max 128 characters".to_string());
+    }
+
+    if s.chars()
+        .any(|c| c.is_ascii() && c.to_ascii_lowercase() >= 'h' && c.to_ascii_lowercase() <= 'z')
+    {
```

Comment:
I don't think this is needed since you're going to return error if `Regex::new()` fails. This will also fail with character classes e.g `--regex "[[:alpha:]]+"`

Reply:
addressed: https://github.com/FuelLabs/sway/pull/6661/commits/5a850d2f485b61a3ef3b1febd1432a9813005253